### ISSUE TITLE
Phone Verification: Refactor ContactInformation Service and API

### ISSUE THREAD

**Issue by nima-karimi** [woocommerce/google-listings-and-ads#981]
As part of #962 to implement the full version of phone verification.

## `ContactInformation` Service

Since after rolling out phone verification, we always verify the user's phone number before setting it in Merchant Center, the methods for setting phone number will become irrelevant. Therefore, this class needs to be refactored so that it only handles updating the store's address.

Given that the phone number returned by Google API doesn't have a specific format, it needs to be normalized into a consistent format to make the `get_contact_information` method's response more deterministic. The `sanitize_phone_number` method can be reused for this purpose.

Note that phone verification is handled by the `PhoneVerification` service: https://github.com/woocommerce/google-listings-and-ads/issues/980

## Contact Information API

The edit contact information API also should be modified so that it doesn't accept a `phone_number` input and only sets the address to the WooCommerce store address once called.

The get contact information API, however, still returns both the phone number and store's address since they are used in the UI.

**Comment by eason9487** [comment]
There're some logics that need the data of whether the saved phone number is verified. Would it be added to the current `mc/contact-information` API or the new API? And I was thinking the phone number returned from the Google side might be in a different data structure when verified and unverified.

**Comment by nima-karimi** [comment]
> I was thinking the phone number returned from the Google side might be in a different data structure when verified and unverified.

Good point 💯 It'd be good to normalize the phone number to a consistent format. Added a note to the issue.

> There're some logics that need the data of whether the saved phone number is verified. Would it be added to the current mc/contact-information API or the new API?

The Account Business Information API doesn't indicate whether a phone number is verified or not. It just returns the phone number itself. And the newly introduced phone number verification APIs do not provide a way to inquire about the phone number's verification status. I went through the Google API docs but couldn't find an API for this.
